resources/aws_ec2.go

```diff
@@ -0,0 +1,535 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	RegisterResource("aws:ec2", func() Res { return &AwsEc2Res{} })
+}
+// AwsPrefix is a const which gets prepended onto the instance name.
+const AwsPrefix = "mgmt:"
+
+// AwsRegions is a list of all AWS regions generated using ec2.DescribeRegions.
```

Comment:
Optional: Add an http link to this comment so users can see these in their browser or figure out easily what's new/changed and write patches?